chain/network/src/peer_manager.rs

```diff
@@ -228,7 +231,24 @@ impl PeerManagerActor {
         });
     }
 
-    fn remove_active_peer(&mut self, ctx: &mut Context<Self>, peer_id: &PeerId) {
+    /// Remove peer from active set.
+    /// Check it match peer_type to avoid removing a peer that both started connection to each other.
+    /// If peer_type is None, remove anyway disregarding who started the connection.
+    fn remove_active_peer(
+        &mut self,
+        ctx: &mut Context<Self>,
+        peer_id: &PeerId,
+        peer_type: Option<PeerType>,
```

Comment:
why can a peer type be `None`?

Reply:
It is in a comment above. When the peer is banned we don't care if the type of connection, we remove it anyway. Otherwise we only remove it if it matches peer_type.

Reply:
I don't think that answers my question. Being `None` means that there is something besides inbound or outbound connection, what is it?

Reply:
There is no other option than Inbound and Outbound. The problem is that when we are banning a peer we might not know if it is inbound or outbound, we don't care, we are banning him anyway, disregarding its type. So basically None tells you to remove it, otherwise, before removing it, check that its type match `Some(peer_type)` so we don't remove the incorrect connection.

This is handling a tricky edge case.

Reply:
Okay I see